<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>08_promise的几个关键问题3</title>
</head>
<body>
<!--
1.
async修饰的函数
函数的返回值为promise.对象
Promise实例的结果由async函数执行的返回值决定
2,await表达式
await右侧的表达式一般为Promise实例对象，但也可以是其它的值
如果表达式是promise对象，await后的返回值是promise成功的值
如果表达式是其它值，直接将此值作为await的返回值
3,注意：
await必须写在async函数中，但async函数中可以没有await
如果await的promise失败了，就会抛出异常，需要通过try.,catch来捕获处理

若我们使用async配合await这种写法：
1,表面上不出现任何的回调函数
2.但实际上底层把我们写的代码进行了加工，把回调函数“还原”回来了。
3,最终运行的代码是依然有回调的，只是程序员没有看见。
-->
<script type="text/javascript">
    const p = new Promise((resolve, reject) => {
        setTimeout(() => {
            resolve()
        }, 1000)
    })
    // p.then(value => console.log(value),
    //     reason => console.log(reason)

    // )
    // async function demo1(){
    //     const result = await p
    //     console.log(result)
    // }

    async function demo() {
        p.then(value => {
            console.log(value)
            console.log(100)
            console.log(200)
        })


    demo()
    console.log(1);
</script>
</body>
</html>